### DISCUSSION TITLE
There appears to be no way to catch certain errors in a schema.

### DISCUSSION THREAD

**Discussion by trusktr** [json-schema-org/community#92] [Q&A]
For example, take this schema:

```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "description": "JSON Schema for representing a Planning Project",
  "type": "object",
  "required": [
    "activityPlan"
  ],
  "properties": {
    "activityPlan": {
      "type": "object",
      "properties": {
        "actions": {
          "type": "array",
          "items": {
            "required": [
              "uuid",
              "actionType",
              "actionStart",
              "actionEnd"
            ],
            "name": {
              "type": "string"
            },
            "uuid": {
              "type": "string",
              "format": "uuid"
            },
            "actionType": {
              "type": "string"
            },
            "actonStart": {
              "type": "string",
              "format": "date-time"
            },
            "actionEnd": {
              "type": "string",
              "format": "date-time"
            }
          }
        }
      }
    }
  }
}
```

There is a typo: `actonStart` should be `actionStart`. 

I can't seem to find any online validators that catch this error. Is there a way to catch this error?

**Comment by karenetheridge** [comment]
How is the validator supposed to know that you spelled a property name wrong?

**Reply by trusktr** [reply]
Yeah, good question. This results in an `any` type for `actionStart` in TypeScript, which should ideally be an error.

**Reply by karenetheridge** [reply]
Are you referring to a specific tool that gives special meaning to these properties?

You can of course add `additionalProperties: false` to the schema, which will give you a validation failure when you receive data containing `"actionStart": ...` rather than `"actonStart": ...`, but then it would be on you to figure out that it's actually an error in your schema and not your data.

**Reply by trusktr** [reply]
Hmmm, thanks for sharing that. I think that would at least be better, because then the error may be found sooner as opposed to on accident while I was perusing.
